salt/_modules/metalk8s_solutions.py

```diff
@@ -173,6 +173,16 @@ def _is_solution_mount(mount_tuple):
     if mount_info["fstype"] != "iso9660":
         return False
 
+    try:
+        read_solution_manifest(mountpoint)
+    except CommandExecutionError as exc:
```

Comment:
Not sure we want to do this `try`/`except` in `_is_solution_mount`. Since we only use this method in `list_available`:

```python
solution_mounts = filter(_is_solution_mount, active_mounts.items())

for mountpoint, mount_info in solution_mounts.items():
    manifest, info = read_solution_manifest(mountpoint)
```

Maybe we could move this try/except to `list_available`, and keep the resulting manifest around if there is one. This will avoid calling `read_solution_manifest` twice.


```python
candidate_mounts = filter(_is_candidate_mount, active_mounts.items())

for mountpoint, mount_info in candidate_mounts.items():
	try:
	    manifest, info = read_solution_manifest(mountpoint)
	except CommandExecutionError as exc:
	    log.info("Skipping ...")
	    continue

    # the rest is unchanged
```

Reply:
Then the `_is_solution_mount` should be renamed.

To me, with this approach, we can easily validate that a mountpoint is a solution just by calling this `_is_solution_mount`

Reply:
Yeah we could do this as well, I'm fine with getting rid of the `filter` call. It's just there to make it clear what objects we want to iterate on, and reduce the clutter inside the `for` loop.

Reply:
Done